cvmfs/download.cc

```diff
@@ -1955,6 +1979,87 @@ void DownloadManager::ProbeHosts() {
 }
 
 
+bool DownloadManager::GeoSortServers(std::vector<std::string> *servers,
+                    std::vector<uint64_t> *output_order) {
+  if (!servers) {return false;}
+  if (servers->size() == 1) {
+    if (output_order) {
+      output_order->clear();
+      output_order->push_back(0);
+    }
+    return true;
+  }
+
+  std::vector<std::string> host_chain;
+  GetHostInfo(&host_chain, NULL, NULL);
+
+  std::vector<std::string> server_dns_names;
+  server_dns_names.reserve(servers->size());
+  for (unsigned i = 0; i < servers->size(); ++i) {
+    std::string host = dns::ExtractHost((*servers)[i]);
+    server_dns_names.push_back(host.empty() ? (*servers)[i] : host);
```

Comment:
Currently merging, I'm wondering if `host.empty() ?` is necessary.  Can we just push back the result from `ExtractHost`?